astarte/device/interface.py

```diff
@@ -188,9 +207,94 @@ def get_reliability(self, endpoint: str) -> int:
             return mapping.reliability
         return 2
 
-    def validate(self, path: str, payload, timestamp: datetime) -> ValidationError | None:
+    def validate_path(self, path: str, payload) -> ValidationError | None:
         """
-        Interface Data validation
+        Validate that the provided path conforms to the interface.
+
+        Parameters
+        ----------
+        path: str
+            Path to validate. In case of an individual interface it should correspond to the full
+            endpoint, while in case of aggregated interfaces it should correspond to the common
+            part to all the endpoints.
+        payload: object
+            Payload used to extrapolate the remaining endpoints for aggregated interfaces.
+
+        Returns
+        -------
+        ValidationError or None
+            None in case of successful validation, ValidationError otherwise
+        """
+        if not self.is_aggregation_object():
+            if not self.get_mapping(path):
+                return ValidationError(f"Path {path} not in the {self.name} interface.")
+        else:
+            for k in payload:
+                if not self.get_mapping(f"{path}/{k}"):
+                    return ValidationError(f"Path {path}/{k} not in the {self.name} interface.")
+        return None
+
+    def validate_payload(self, path: str, payload) -> ValidationError | None:
+        """
+        Validate that the payload conforms to the interface.
+
+        Parameters
+        ----------
+        path: str
+            Path on which the payload has been received. This is assumed to correspond to a valid
+            mapping (or partial mapping in case of aggregate interface). Should be first checked
+            with validate_path().
+        payload: object
+            Data to validate
+
+        Returns
+        -------
+        ValidationError or None
+            None in case of successful validation, ValidationError otherwise
+        """
+        if not self.is_aggregation_object():
+            return self.get_mapping(path).validate_payload(payload)
+        if not isinstance(payload, dict):
+            return ValidationError(f"Payload not a dict for aggregated interface {self.name}.")
+        for k, v in payload.items():
+            payload_invalid = self.get_mapping(f"{path}/{k}").validate_payload(v)
+            if payload_invalid:
+                return payload_invalid
+        # Check all the interface endpoints are present in the payload
+        return self.validate_aggregate_complete(path, payload)
+
+    def validate_aggregate_complete(self, path: str, payload):
```

Comment:
This seems to be the validation for the object more than a generic aggreagate. Could we change the name in validate_aggregate_object or validate_object_complete?